d2bs/kolbot/libs/bots/TeamDiablo.js

```diff
@@ -0,0 +1,462 @@
+                    this.common.doSeal(who);
+                }
+            }
+
+            // In the you define teams to do the seals, but a teammember is missing, make sure we did all the seals
+            if (Config.TeamDiablo.Viz || Config.TeamDiablo.Seis || Config.TeamDiablo.Inf) {
+
+                // Either one is defined we should do. So make sure we did all. Or help others
+                while (!Team.communication.diablo.vizDone || !Team.communication.diablo.seisDone || !Team.communication.diablo.vizDone) {
+
+                    for (who in diablo.data.seals) {
+                        if (!Team.communication.diablo[who + 'Done'] && !Team.communication.diablo.diabloDone) {
+                            this.common.doSeal(who);
+                        }
+                    }
+                }
+            }
+
+            if (Config.TeamDiablo.Diablo && !Team.communication.diablo.diabloDone) {
+                this.common.killDia();
+            }
+
+            if (!Team.communication.diablo.diabloDone) {
+                Town.goToTown(4);
+
+                while (!baal.data.communication.diabloDone) {
+                    // ToDo: Should make this more fail proof
+                    delay(1000);
+                }
+            }
+        },
+        common: {
+            teleport: function () {
+                diablo.util.print('Teleporting to chaos');
+                Town.doChores();
+
+
+                Pather.useWaypoint(Areas.RiverOfFlame, true);
+                Precast.doPrecast(true);
+
+                // Doing chores & precast can take some time. So much time in fact, maybe another char already is at chaos, or almost
+                if (Team.communication.diablo.portalReady || Team.communication.diablo.atChaos) {
+                    return Pather.useWaypoint(Areas.PandemoniumFortress); // Returning true, since we succeed as a team already to go to choas
+                }
+
+                // Move to Chaos
+                if (!Pather.moveToExit(Areas.ChaosSanctuary, false)) {
+                    diablo.util.print("Failed to move to chaos sanctuary.");
+                    return false;
+                }
+
+                // Let others know this char is at chaos
+                Team.communication.diablo.atChaos = true;
+                Team.msg.sendData('diablo');
+
+
+                Pather.moveTo(diablo.data.portalSpot[0], diablo.data.portalSpot[1]);
+
+                // Cast tp in ase there is no portal up yet
+                if (!Pather.getPortal(Areas.PandemoniumFortress)) {
+                    Pather.makePortal();
+                    Team.communication.diablo.portalReady = true;
+                    Team.msg.sendData('diablo');
+                }
+                return true;
+            },
+            doSeal: function (who) {
+                if (Team.communication.diablo[who + 'Done']) {
+                    return true;
+                }
+                if (!diablo.util.toChaos()) {
```

Comment:
not sure why 'teleport' is in common but 'toChaos' is in util, don't think the sub class designations are useful